components/camel-ai/camel-openai/src/main/docs/openai-component.adoc

```diff
@@ -0,0 +1,358 @@
+= OpenAI Component
+:doctitle: OpenAI
+:shortname: openai
+:artifactid: camel-openai
+:description: OpenAI endpoint for chat completion.
+:since: 4.17
+:supportlevel: Preview
+:tabs-sync-option:
+:component-header: Only producer is supported
+
+*Since Camel {since}*
+
+*{component-header}*
+
+The OpenAI component provides integration with OpenAI and OpenAI-compatible APIs for chat completion using the official openai-java SDK.
+
+== URI Format
+
+[source]
+----
+openai:operation[?options]
+----
+
+Currently, only the `chat-completion` operation is supported.
+
+== Component Options
```

Comment:
These should be generated, like the partials in https://github.com/apache/camel/blob/main/components/camel-ai/camel-langchain4j-agent/src/main/docs/langchain4j-agent-component.adoc#uri-format